runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -629,12 +625,14 @@ impl<T: Trait> Module<T> {
     // Enumerates through active proposals. Tally Voting results.
     // Returns proposals with finalized status and id
     fn get_finalized_proposals() -> Vec<FinalizedProposal<T>> {
-        // Enumerate active proposals id and gather finalization data.
+        // Filter active proposals and gather finalization data.
         // Skip proposals with unfinished voting.
-        <ActiveProposalIds<T>>::iter()
-            .filter_map(|(proposal_id, _)| {
-                // load current proposal
-                let proposal = Self::proposals(proposal_id);
+        <Proposals<T>>::iter()
+            .filter_map(|(proposal_id, proposal)| {
+                // Filter only active proposals.
+                if !proposal.status.is_active_proposal() {
```

Comment:
This segment can be made clearer. 

A proposal is either active, or it is not, in which case there is a decision status for it. This code however does not statically capture this, it appears as if `decision_status` could theoretically be `None`, but this will never happen at runtime. This can be improved to statically capture this, raising clarity and safety when someone tries to update this in the future. Comment should also be changed.

Reply:
Ok. I implemented the suggested dichotomy (active/others) using `match proposal.status` and commented 'others' branch.